examples/direct_fidelity_estimation.py

```diff
@@ -370,11 +380,16 @@ def direct_fidelity_estimation(circuit: cirq.Circuit, qubits: List[cirq.Qid],
             noisy_simulator.simulate(circuit)).final_density_matrix
 
     trial_results: List[TrialResult] = []
-    for _ in range(n_trials):
-        # Randomly sample as per probability.
-        i = np.random.choice(len(pauli_traces), p=p)
+    for i_seq in range(len(pauli_traces)):
```

Comment:
not blocking for this PR but keep in mind:

I still feel like this could be structured in a more readable ("pythonic") way. So you have `pauli_traces` and you need to do some of them with some probability that depends on which modality you're operating in. Why not factor the generation of the ones you want to measure outside the loop?

```python
if clifford etc:
  measured_ptraces = pauli_traces
else:
  measure_ptraces = np.choice(pauli_traces, size=len(pauli_traces))

for mpt in measure_ptraces:
  do_jazzy_stuff(mpt.P_i)
  do_more_jazzy_stuff(mpt.rho_i)
  trial_results.append(TrialResult(sigma_i=sigma_i))
```

Every time I see an `i` index in a Python for loop, I get very suspicious.

Related question: what is a user supposed to do with the `i` that you save in the `TrialResult`? is that helpful? Maybe just give them the whole pauli_trace instead of making them have to look it up?

Reply:
Done.